nginx-create-session-ticket-keys

```diff
@@ -1,15 +1,12 @@
-#!/bin/bash
+#!/bin/sh
 
-set -o errexit -o nounset -o pipefail
+set -o errexit -o nounset
 
 umask 077
 
 mkdir -p /etc/nginx/session-ticket-keys
 mount -t ramfs -o mode=700 ramfs /etc/nginx/session-ticket-keys
 
-cd /etc/nginx/session-ticket-keys
-
-openssl rand -out 1.key 80
-openssl rand -out 2.key 80
-openssl rand -out 3.key 80
-openssl rand -out 4.key 80
+for i in 1 2 3 4; do
+	dd ibs=1 obs=1 count='80' 2>/dev/null </dev/urandom >"/etc/nginx/session-ticket-keys/$i.key"
```

Comment:
urandom on Linux still doesn't wait for initialization of the CSPRNG on boot which is a real problem for this. /dev/random is fine on modern kernels and matches `getrandom` but on older ones has a totally nonsense design instead of simply blocking once at boot.

Reply:
Are there any disadvantages to using `/dev/random` instead of `/dev/urandom`, assuming we aren't using an old kernel?

Reply:
Using openssl should use `getentropy` when available with fallback to `/dev/urandom` when it isn't which is why I used it. I'd be fine with hard-wiring it to use `/dev/random` and ignoring old kernels. I don't see much point in avoiding OpenSSL because nginx depends on having OpenSSL, LibreSSL or BoringSSL.

Reply:
i.e. this isn't adding any extra dependency on top of nginx and isn't trusting something beyond what nginx already trusts to be done properly.

Reply:
Nginx depends on \*ssl libraries, but not the commandline executable. Personally, I build nginx against boringssl but my distros only packages openssl.

I get that this isn't an issue for GrapheneOS servers, but if there's no downside to using `/dev/random` instead then we can use that instead. I think it's fine to ignore servers with older kernels since their sysadmins should probably address their unpatched vulns before worrying about something  like session tickets.

Reply:
The kernel.org LTS kernel branches are supported by 6 years now so there are a bunch of LTS branches that are still fully supported without the sane `/dev/random` semantics since that change was quite recent.